doc/release-process.md

```diff
@@ -1,14 +1,54 @@
 # Release Process
 
-1. Open PR to master that
-   1. adds release notes to `doc/CHANGELOG.md` and
-   2. if this is **not** a patch release, updates `_PKG_VERSION_{MAJOR,MINOR}` and `_LIB_VERSIONS_*` in `configure.ac`
-2. After the PR is merged,
-   * if this is **not** a patch release, create a release branch with name `MAJOR.MINOR`.
-     Make sure that the branch contains the right commits.
-     Create commit on the release branch that sets `_PKG_VERSION_IS_RELEASE` in `configure.ac` to `true`.
-   * if this **is** a patch release, open a pull request with the bugfixes to the `MAJOR.MINOR` branch.
-     Also include the release note commit bump `_PKG_VERSION_BUILD` and `_LIB_VERSIONS_*` in `configure.ac`.
-4. Tag the commit with `git tag -s vMAJOR.MINOR.PATCH`.
-5. Push branch and tag with `git push origin --tags`.
-6. Create a new GitHub release with a link to the corresponding entry in `doc/CHANGELOG.md`.
+This document outlines the process for releasing versions of the form `$MAJOR.$MINOR.$PATCH`.
+
+We distinguish between two types of releases: *regular* and *maintenance* releases.
+Regular releases are releases of a new major or minor version as well as patches of the most recent release.
+Maintenance releases, on the other hand, are required for patches of older releases.
+
+Note that adding a release entry to [CHANGELOG.md](../CHANGELOG.md) requires a release date. This date should match the dates of the remaining steps in the release process, including the date of the tag and the GitHub release.
```

Comment:
Is ti worth mentioning in the `CHANGELOG.md` for each regular release whether it is ABI backward compatible or not?
A clearl statement about ABI backward compatibility will also help to modify `_LIB_VERSION_*` variables properly.

Reply:
It would make more sense to me to do the opposite and mention that we make little guarantees about the library version. We don't have methods to accurately determine ABI compatibility and so we can either put reasonable effort into attempting proper library versioning or not having library versions at all. So far the consensus seemed to be that we do the former.

Reply:
OTOH if our resonable-effort approach to library versioning would indicate that it's not consistent with semantic versioning, then I agree that it makes sense to mention that in the changelog. You think that's worth adding to the `release-process.md`?

Reply:
> I don't think it's unreasonable to at least try and have a very good chance of getting it right

Good, then there doesn't seem to be disagreement.

If we want to mention ABI compatibility in every  release note, we could add a section `### ABI compatibility` (or so) to the release notes that (somewhat cautiously) mentions how we set the library version.

Reply:
I added a section `### ABI compatibility` to the changelog and mention it in the release process.